Makefile

```diff
@@ -40,8 +89,25 @@ test-coverage:
 	echo "" > $(COVERAGE_REPORT); \
 	$(GOTEST) -coverprofile=$(COVERAGE_REPORT) -coverpkg=./... -covermode=$(COVERAGE_MODE) ./...
 
+.PHONY: clean clean-docker clean-all
+
 clean:
-	rm -rf $(GIT_DIST_PATH)
+	@if [ -d $(GIT_DIST_PATH) ]; then
+		rm -rf $(GIT_DIST_PATH)
+	fi
+	if [ -f $(TEST_ENV) ] || [ -f $(DOCKER_ENV) ]; then
+		rm $(TEST_ENV) $(DOCKER_ENV)
+	fi
+	rm coverage*
```

Comment:
We could keep it as it was, but instead have a single dir (e.g. `build/`) in which any ephemeral things are created into. So we can remove a single dir to clear up any build/test artefacts.